src/pykeen/nn/modules.py

```diff
@@ -1409,6 +1409,44 @@ def _prepare_hrt_for_functional(
         return dict(h=h, r_h=r[0], r_t=r[1], t=t)
 
 
+def quaterion_multiplication_table() -> torch.Tensor:
```

Comment:
Why not just save the result if this function to a global variable? Are there reasons to use alternative tables?

Reply:
I was wondering whether there would be problems with the interaction with [`torch.set_default_dtype`](https://pytorch.org/docs/stable/generated/torch.set_default_dtype.html), so I decided to instantiate the tensor at runtime, rather than at import-time.

Reply:
okay this is fine for now, but a little confusing since there aren't any alternative tables used anywhere else at the moment

Reply:
It we were to implement OctonionE (which was proposed in the same paper, but had similar performance while having more parameters), we could to that be an additional table, cf. https://en.wikipedia.org/wiki/Octonion#Definition

Reply:
Okay cool

Reply:
let's move this to its own module then, since the obvious next step is to add the octonion multiplication table

Reply:
https://github.com/pykeen/pykeen/compare/e07c90e4ab7f692be74a04dec17de06f3eb07a0d...55bafdb1690dc430320840d09d41f4f0026d64ff